### PR TITLE
Bumped Nuke to v9

### PR DESCRIPTION
**Pull request by dennisdoomen**
Fixed some of the breaking changes. 

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [ ] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [ ] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -9,19 +9,15 @@ SCRIPT_DIR=$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)
 # CONFIGURATION
 ###########################################################################
 
-BUILD_PROJECT_FILE="$SCRIPT_DIR/Build/_build.csproj"
-TEMP_DIRECTORY="$SCRIPT_DIR/.nuke/temp"
+BUILD_PROJECT_FILE="$SCRIPT_DIR/build/_build.csproj"
+TEMP_DIRECTORY="$SCRIPT_DIR//.nuke/temp"
```
[the remark this answers is not in the document]
Comment (dennisdoomen) [reply build.sh:13]: I tried `nuke :upgrade`, but that seems to break our specific modifications.

File: Build/_build.csproj [hunk 2]
```diff
@@ -22,7 +23,7 @@
     <PackageDownload Include="ReportGenerator" Version="[5.2.0]" />
     <PackageDownload Include="xunit.runner.console" Version="[2.9.2]" />
     <PackageReference Include="LibGit2Sharp" Version="0.30.0" />
-    <PackageReference Include="Nuke.Common" Version="$(NukeVersion)" />
+    <PackageReference Include="Nuke.Common" Version="9.0.2" />
     <PackageReference Include="Nuke.Components" Version="$(NukeVersion)" />
```
[the remark this answers is not in the document]
Comment (dennisdoomen) [reply Build/_build.csproj:27]: I tried nuke :upgrade, but that seems to break our specific modifications.

File: build.ps1 [hunk 3]
```diff
@@ -14,18 +14,14 @@ $PSScriptRoot = Split-Path $MyInvocation.MyCommand.Path -Parent
...
 $DotNetInstallUrl = "https://dot.net/v1/dotnet-install.ps1"
-$DotNetChannel = "Current"
+$DotNetChannel = "STS"
 
-$env:DOTNET_SKIP_FIRST_TIME_EXPERIENCE = 1
 $env:DOTNET_CLI_TELEMETRY_OPTOUT = 1
-$env:DOTNET_MULTILEVEL_LOOKUP = 0
-$env:DOTNET_ROLL_FORWARD = "Major"
-$env:NUKE_TELEMETRY_OPTOUT = 1
-$env:DOTNET_CLI_UI_LANGUAGE = "en-US"
```
[the remark this answers is not in the document]
Comment (dennisdoomen) [reply build.ps1:28]: I tried nuke :upgrade, but that seems to break our specific modifications.

**Comment by ITaluone** [comment]
@dennisdoomen Do you know why `UnitTestsNet47` are working here, but not here: https://github.com/fluentassertions/fluentassertions/pull/2861 ?

**Comment by dennisdoomen** [comment]
> Do you know why `UnitTestsNet47` are working here, but not here: #2861 ?

There was 9.0.2 release last night. Maybe something was broken there?

**Comment by ITaluone** [comment]
Hmm.. I checked the commit log.. no commit changed something in the area I expected a change.. I'll try locally...

**Pull request merged**